src/include/platform/ICDManager.h

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+namespace chip {
+namespace DeviceLayer {
+
+/**
+ * @brief ICD Manager is responsible of processing the events and triggering the correct action for an ICD
```

Comment:
```suggestion
 * @brief ICD Manager is responsible for doing application-specific processing of ICD-related notifications from the SDK.
```

Or at least I would think that's how it should work: the SDK should do all the spec-required bits it knows how to do, and call out to the ICDManager for anything it needs from the application.

Importantly: it looks like ICDManager is public SDK API while ICDEventManager is an SDK implementation detail, and arguably one that should not live under `platform` at all....  But it's hard to tell for sure if all we have are stubs and there is no documentation or examples that hints at what those stubs will really do.